dom-event-delegation/main.ts

```diff
@@ -0,0 +1,18 @@
+const $taskList = document.querySelector('.task-list');
+if (!$taskList) {
+  throw new Error('$taskList not found!');
+}
+$taskList.addEventListener('click', clickHandler);
+
+function clickHandler(event: Event): void {
+  const eventTarget = event.target as HTMLElement;
+  console.log('eventTarget variable:', eventTarget);
+  console.log('eventTarget tagName:', eventTarget.tagName);
+  if (eventTarget.tagName === 'BUTTON') {
+    const $targetTaskListItem = eventTarget.closest(
+      '.task-list-item'
+    ) as HTMLElement;
+    console.log("eventTarget's nearest li ancestor:", $targetTaskListItem);
+    $targetTaskListItem.remove();
+  }
```

Comment:
When using event delegation, I like to use "guard clauses." This means that if something is _not_ true, then I return from the function and don't look at the rest of the code. It makes it _slightly_ (probably by a minuscule amount) more efficient but helps with readability.

```
if ($eventTarget.tagName !== 'BUTTON') {
    return;
  }

  const $taskListItem = $eventTarget.closest('.task-list-item');

  console.log('closest .task-list-item', $taskListItem);
  $taskListItem?.remove();
```